src/simple_key_provider.rs

```diff
@@ -17,6 +17,67 @@ pub struct SimpleKeyProvider<R: SeedableRng + RngCore = ChaChaRng> {
     rng: RefCell<R>,
 }
 
+// EncryptedSimpleKey relies on this size being constant. If this ever needs to be changed a new
+// version of EncryptedSimpleKey needs to be created.
+type Nonce = aes_gcm::Nonce<U12>;
+const NONCE_SIZE: usize = 12;
+
+/// A decoded intermediate representation of an encrypted simple key
+///
+/// The data this struct is created from exists in memory like so:
```

Comment:
Both I suppose. This struct doesn't "own" the data and can be built from a contiguous region of memory - or it can be built from separate slices and serialised to a contiguous vec. I'll try and rework this comment.